====Input Arguments====
{
    "data_path": "./dataset/test/medqa.jsonl",
    "output_path": "./results/medqa/medqa_5op-llama3-8b-sft",
    "chunk_num": 1,
    "chunk_idx": 0,
    "resume": true,
    "model_name_or_path": "/mnt/hwfile/medai/LLMModels/Model/Meta-Llama-3-8B",
    "peft_path": "/mnt/hwfile/medai/liaoyusheng/projects/LLM-REASONING/DataSeletion/checkpoints/medqa_5op-llama3-8b-sft",
    "use_vllm": false,
    "gpu_memory_utilization": 0.7,
    "prepare_func": null,
    "batch_size": 4,
    "temperature": 1.0,
    "max_new_tokens": 2048,
    "direct_answer": false,
    "cache_file": "cache.jsonl",
    "result_file": "result.json"
}
Loading checkpoint shards:   0%|          | 0/4 [00:00<?, ?it/s]Loading checkpoint shards:  25%|██▌       | 1/4 [00:00<00:02,  1.30it/s]Loading checkpoint shards:  50%|█████     | 2/4 [00:01<00:01,  1.16it/s]Loading checkpoint shards:  75%|███████▌  | 3/4 [00:02<00:00,  1.20it/s]Loading checkpoint shards: 100%|██████████| 4/4 [00:02<00:00,  1.51it/s]Loading checkpoint shards: 100%|██████████| 4/4 [00:02<00:00,  1.38it/s]
/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:166: UserWarning: Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes


  warn(msg)
/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:166: UserWarning: /mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:166: UserWarning: Found duplicate ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] files: {PosixPath('/mnt/lustre/share/cuda-11.8/lib64/libcudart.so.11.0'), PosixPath('/mnt/lustre/share/cuda-11.8/lib64/libcudart.so')}.. We select the PyTorch default libcudart.so, which is {torch.version.cuda},but this might missmatch with the CUDA version that is needed for bitsandbytes.To override this behavior set the BNB_CUDA_VERSION=<version string, e.g. 122> environmental variableFor example, if you want to use the CUDA version 122BNB_CUDA_VERSION=122 python ...OR set the environmental variable in your .bashrc: export BNB_CUDA_VERSION=122In the case of a manual override, make sure you set the LD_LIBRARY_PATH, e.g.export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda-11.2
  warn(msg)
/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:166: UserWarning: /mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/cv2/../../lib64:/mnt/lustre/share/cuda-11.8/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib:/mnt/petrelfs/share/gcc/gmp-4.3.2/lib/:/mnt/petrelfs/share/gcc/mpfr-2.4.2/lib/:/mnt/petrelfs/share/gcc/mpc-0.8.1/lib:/mnt/petrelfs/share/gcc:/mnt/lustre/share/cuda-11.8/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib:/mnt/petrelfs/share/gcc/gmp-4.3.2/lib/:/mnt/petrelfs/share/gcc/mpfr-2.4.2/lib/:/mnt/petrelfs/share/gcc/mpc-0.8.1/lib:/mnt/petrelfs/share/gcc:/mnt/lustre/share/cuda-11.8/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib64:/mnt/petrelfs/share/gcc/gcc-9.3.0/lib:/mnt/petrelfs/share/gcc/gmp-4.3.2/lib/:/mnt/petrelfs/share/gcc/mpfr-2.4.2/lib/:/mnt/petrelfs/share/gcc/mpc-0.8.1/lib:/mnt/petrelfs/share/gcc: did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:166: UserWarning: Found duplicate ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] files: {PosixPath('/usr/local/cuda/lib64/libcudart.so.11.0'), PosixPath('/usr/local/cuda/lib64/libcudart.so')}.. We select the PyTorch default libcudart.so, which is {torch.version.cuda},but this might missmatch with the CUDA version that is needed for bitsandbytes.To override this behavior set the BNB_CUDA_VERSION=<version string, e.g. 122> environmental variableFor example, if you want to use the CUDA version 122BNB_CUDA_VERSION=122 python ...OR set the environmental variable in your .bashrc: export BNB_CUDA_VERSION=122In the case of a manual override, make sure you set the LD_LIBRARY_PATH, e.g.export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda-11.2
  warn(msg)
False

===================================BUG REPORT===================================
================================================================================
The following directories listed in your path were found to be non-existent: {PosixPath('/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/cv2/../../lib64')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('file'), PosixPath('/mnt/petrelfs/liaoyusheng/miniconda3/etc/xml/catalog file'), PosixPath('/etc/xml/catalog file')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('/mnt/petrelfs/liaoyusheng/.vscode-server/extensions/ms-python.debugpy-2025.5.2025031101-linux-x64/.noConfigDebugAdapterEndpoints/endpoint-a33c6d756027547a.txt')}
The following directories listed in your path were found to be non-existent: {PosixPath('vpI7guUDtAFVBtfnw5DKTEgny366SwFk39GSL32PTjA1kplMOVs2TyKJ6ChS@10.1.20.50'), PosixPath('23128'), PosixPath('//liaoyusheng'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('//closeai-proxy.pjlab.org.cn'), PosixPath('23128'), PosixPath('http')}
The following directories listed in your path were found to be non-existent: {PosixPath('/run/user/200002137/vscode-git-995625de8e.sock')}
The following directories listed in your path were found to be non-existent: {PosixPath('/run/user/200002137/vscode-ipc-397f4a0a-1584-495e-91a5-dbd2f0c1764e.sock')}
The following directories listed in your path were found to be non-existent: {PosixPath('https'), PosixPath('//hf-mirror.com')}
The following directories listed in your path were found to be non-existent: {PosixPath('() {  eval `/usr/bin/modulecmd bash $*`\n}')}
CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. Searching in backup paths...
DEBUG: Possible options found for libcudart.so: {PosixPath('/usr/local/cuda/lib64/libcudart.so.11.0'), PosixPath('/usr/local/cuda/lib64/libcudart.so')}
CUDA SETUP: PyTorch settings found: CUDA_VERSION=124, Highest Compute Capability: 8.0.
CUDA SETUP: To manually override the PyTorch CUDA version please see:https://github.com/TimDettmers/bitsandbytes/blob/main/how_to_use_nonpytorch_cuda.md
CUDA SETUP: Required library version not found: libbitsandbytes_cuda124.so. Maybe you need to compile it from source?
CUDA SETUP: Defaulting to libbitsandbytes_cpu.so...

================================================ERROR=====================================
CUDA SETUP: CUDA detection failed! Possible reasons:
1. You need to manually override the PyTorch CUDA version. Please see: "https://github.com/TimDettmers/bitsandbytes/blob/main/how_to_use_nonpytorch_cuda.md
2. CUDA driver not installed
3. CUDA not installed
4. You have multiple conflicting CUDA libraries
5. Required library not pre-compiled for this bitsandbytes release!
CUDA SETUP: If you compiled from source, try again with `make CUDA_VERSION=DETECTED_CUDA_VERSION` for example, `make CUDA_VERSION=113`.
CUDA SETUP: The CUDA version for the compile might depend on your conda install. Inspect CUDA version via `conda list | grep cuda`.
================================================================================

CUDA SETUP: Something unexpected happened. Please compile from source:
git clone https://github.com/TimDettmers/bitsandbytes.git
cd bitsandbytes
CUDA_VERSION=124
python setup.py install
CUDA SETUP: Setup Failed!
Traceback (most recent call last):
  File "/mnt/hwfile/medai/liaoyusheng/projects/LLM-REASONING/DataSeletion/DataSelection0326/eval_cot/eval_cot.py", line 207, in <module>
    main()
  File "/mnt/hwfile/medai/liaoyusheng/projects/LLM-REASONING/DataSeletion/DataSelection0326/eval_cot/eval_cot.py", line 196, in main
    model, tokenizer = load_model_and_tokenizer(args.model_name_or_path, args.peft_path)
  File "/mnt/hwfile/medai/liaoyusheng/projects/LLM-REASONING/DataSeletion/DataSelection0326/eval_cot/eval_cot.py", line 183, in load_model_and_tokenizer
    model = PeftModel.from_pretrained(model, peft_path)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/peft_model.py", line 352, in from_pretrained
    model = MODEL_TYPE_TO_PEFT_MODEL_MAPPING[config.task_type](model, config, adapter_name)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/peft_model.py", line 1059, in __init__
    super().__init__(model, peft_config, adapter_name)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/peft_model.py", line 126, in __init__
    self.base_model = cls(model, {adapter_name: peft_config}, adapter_name)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/lora/model.py", line 111, in __init__
    super().__init__(model, config, adapter_name)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/tuners_utils.py", line 147, in __init__
    self.inject_adapter(self.model, adapter_name)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/tuners_utils.py", line 302, in inject_adapter
    self._create_and_replace(peft_config, adapter_name, target, target_name, parent, current_key=key)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/lora/model.py", line 182, in _create_and_replace
    new_module = self._create_new_module(lora_config, adapter_name, target, **kwargs)
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/lora/model.py", line 244, in _create_new_module
    from .bnb import dispatch_bnb_8bit
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/peft/tuners/lora/bnb.py", line 18, in <module>
    import bitsandbytes as bnb
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/__init__.py", line 6, in <module>
    from . import cuda_setup, utils, research
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/research/__init__.py", line 1, in <module>
    from . import nn
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/research/nn/__init__.py", line 1, in <module>
    from .modules import LinearFP8Mixed, LinearFP8Global
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/research/nn/modules.py", line 8, in <module>
    from bitsandbytes.optim import GlobalOptimManager
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/optim/__init__.py", line 6, in <module>
    from bitsandbytes.cextension import COMPILED_WITH_CUDA
  File "/mnt/petrelfs/liaoyusheng/miniconda3/envs/reasoning/lib/python3.10/site-packages/bitsandbytes/cextension.py", line 20, in <module>
    raise RuntimeError('''
RuntimeError: 
        CUDA Setup failed despite GPU being available. Please run the following command to get more information:

        python -m bitsandbytes

        Inspect the output of the command and see if you can locate CUDA libraries. You might need to add them
        to your LD_LIBRARY_PATH. If you suspect a bug, please take the information from python -m bitsandbytes
        and open an issue at: https://github.com/TimDettmers/bitsandbytes/issues
srun: error: SH-IDC1-10-140-0-170: task 0: Exited with exit code 1
